apps/quickbooks_online/views.py

```
@@ -78,7 +79,8 @@ def post(self, request, *args, **kwargs):
 
         # Check for a valid workspace and qbo creds and respond with 400 if not found
         Workspace.objects.get(id=kwargs['workspace_id'])
-        QBOCredential.get_active_qbo_credentials(kwargs['workspace_id'])
+        quickbooks_credentials = QBOCredential.get_active_qbo_credentials(kwargs['workspace_id'])
+        QBOConnector(quickbooks_credentials, workspace_id=kwargs['workspace_id'])
```

this is not needed, UI does a token health call already and that takes care of invalidating

---

I tested this, and we were not reaching the `except:` block of `handle_view_exceptions` when we hit this endpoint. Sync and refresh dimensions were silently failing in case there was an invalid token error

I think `workspaces/:id/qbo/preferences/` is the token health check endpoint. If that is sufficient, should we remove `invalidate_token` from `@handle_view_exceptions`?

---

yes correct, `qbo/preferences/` should do, green to remove invalidate from view decorator

---

removed in https://github.com/fylein/fyle-qbo-api/pull/742/commits/a1f6ef79580196c1659d856114ad681ac1870fce